control/src/http/client_context.h

```diff
@@ -47,8 +47,31 @@ class ClientContext : public ClientContextBase {
     return config_;
   }
 
-  const std::vector<::istio::quota::Requirement>& legacy_quotas() const {
-    return legacy_quotas_;
+  // Append the legacy quotas.
+  void AddLegacyQuotas(std::vector<::istio::quota::Requirement>* quotas) const {
+    quotas->insert(quotas->end(), legacy_quotas_.begin(), legacy_quotas_.end());
+  }
+
+  // Get valid service name in the config map.
+  // If input service name is in the map, use it, otherwise, use the default
+  // one.
+  const std::string& GetServiceName(const std::string& service_name) const {
```

Comment:
Is there ever a case where `GetServiceName` and `GetServiceConfig` would not be used together? As it is right now, this is doing a duplicate map look-up back-to-back which could be combined.

Reply:
The first call result is used for checking the cache map.   If cache hit, the second call is not needed. 
But you are right,  it is two map lookup; first is the service_configs map,  the  second is the cache service_context_map_.  I will optimize it to one lookup.